src/ui/components/results/alternativeverticalscomponent.js

```diff
@@ -53,6 +55,15 @@ export default class AlternativeVerticalsComponent extends Component {
      * @type {boolean}
      */
     this._isShowingResults = opts.isShowingResults || false;
+
+    this.core.globalStorage.on('update', StorageKeys.API_CONTEXT, () => {
+      this.verticalSuggestions = AlternativeVerticalsComponent._buildVerticalSuggestions(
+        this._alternativeVerticals,
+        this._verticalsConfig,
+        this.core.globalStorage.getState(StorageKeys.API_CONTEXT)
+      );
+      this.setState(this.getState());
```

Comment:
Same thing here about what's being passed into setState. I think we need to pass in what's stored in global storage for ALTERNATIVE_VERTICALS. I think this will go for UniversalResults and VerticalResults as well.

Reply:
Changed! Thanks